### PR TITLE
North Carolina Use Tax

### LINKED ISSUE #2761: North Carolina Use Tax
Reference:
https://www.ncdor.gov/taxes-forms/sales-and-use-tax/use-tax
https://www.ncdor.gov/2022-d-400-web-fill-version/open
https://www.ncdor.gov/2022-d-401-individual-income-tax-instructions/open
https://docs.google.com/document/d/1pocvmhC2v5a7vFYG8AALWDbMqFR0EMpd1FlFQqp5xIo/edit

### PR DESCRIPTION
**Pull request by TurnerLuo**
### Summary
📝🛍️🇺🇸

Added a parameter file for North Carolina use tax without record. This file defines the tax rate and threshold for this type of tax, which applies to out-of-state purchases that are taxable in North Carolina. The file is part of the `policyengine_us` project.

> _`parameter` file_
> _defines use tax for state_
> _autumn leaves fall_

### Walkthrough
*  Add a description to the parameter file `without_record.yaml` that explains the use tax rate and threshold for North Carolina taxpayers who do not have records of their out-of-state purchases ([link](https://github.com/PolicyEngine/policyengine-us/pull/2766/files?diff=unified&w=0#diff-dbb04f46990e2a75443f2d44717744cea9343acfc58266cac4d245c0f48bea2dL1-L-1))

fixes #2761

### CODE REVIEW

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: add changelog

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  label: NC Use Tax amount cap for different taxable income
+  reference:
+    - title: NCDOR Use Tax
+      href: https://www.ncdor.gov/taxes-forms/sales-and-use-tax/use-tax
+    - title: 2022 NC D-400
+      href: https://www.ncdor.gov/2022-d-400-web-fill-version/open
+    - title: 2022 D-401 Instruction Booklet
+      href: https://www.ncdor.gov/2022-d-401-individual-income-tax-instructions/open
```
Comment (ThapeloMasasa) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml:13]: add page number for easy reference on document
i.e https://www.ncdor.gov/2022-d-401-individual-income-tax-instructions/open#page=23
This link will take you to excactly page 23 of the document

File: policyengine_us/tests/policy/baseline/gov/states/nc/tax/nc_use_tax.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+- name: test 1
```
Comment (ThapeloMasasa) [inline policyengine_us/tests/policy/baseline/gov/states/nc/tax/nc_use_tax.yaml:1]: Try and be more descriptive in the naming of your test cases
i.e NC use tax for agi of 3500

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - North Carolina Use Tax
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - North Carolina Use Tax.
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,138 @@
+description:  North Carolina Use Tax using this amount for taxpayers without out-of-state purchases records.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml:1]:
```suggestion
description: North Carolina extends the following use tax rate base on taxable income.
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  label: NC Use Tax amount cap for different taxable income
+  reference:
+    - title: NCDOR Use Tax
+      href: https://www.ncdor.gov/taxes-forms/sales-and-use-tax/use-tax
+    - title: 2022 NC D-400
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml:10]:
```suggestion
    - title: 2022 NC D-400 page 2
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+      href: https://www.ncdor.gov/2022-d-400-web-fill-version/open #page=2
+    - title: 2022 D-401 Instruction Booklet
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml:12]:
```suggestion
    - title: 2022 D-401 Instruction Booklet page 23
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+      href: https://www.ncdor.gov/2022-d-401-individual-income-tax-instructions/open #page=23
+brackets:
+  - threshold:
+      2022-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml:16]: need 2021 values

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml [hunk 8]
```diff
@@ -0,0 +1,138 @@
+description:  North Carolina Use Tax using this amount for taxpayers without out-of-state purchases records.
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  label: NC Use Tax amount cap for different taxable income
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml:6]:
```suggestion
  label: North Carolina use tax cap
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/amount.yaml:1]: rename to cap.yaml

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/rate.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
+description:  North Carolina Use Tax using this rate for taxpayers without out-of-state purchases records (taxable income over 45,200).
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/rate.yaml:1]:
```suggestion
description: North Carolina allows for the following use tax rate, based on taxable income.
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/rate.yaml [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  amount_unit: /1
+  label: NC Use Tax rate for taxable income over 45,200
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/rate.yaml:6]:
```suggestion
  label: North Carolina use tax rate
```

File: policyengine_us/variables/gov/states/nc/tax/income/nc_use_tax.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
+from policyengine_us.model_api import *
+
+
+class nc_use_tax(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "NC use tax"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nc/tax/income/nc_use_tax.py:7]:
```suggestion
    label = "North Carolina use tax"
```

File: changelog_entry.yaml [hunk 12]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - North Carolina Use Tax.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - North Carolina use tax.
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/cap.yaml [hunk 13]
```diff
@@ -0,0 +1,138 @@
+description:  North Carolina Use Tax using this cap, based on adjusted gross income.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/cap.yaml:1]:
```suggestion
description: North Carolina limits the use tax to this cap, based on adjusted gross income.
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/cap.yaml [hunk 14]
```diff
@@ -0,0 +1,138 @@
...
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  amount_unit: currency-USD
+  label: North Carolina Use Tax cap
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/cap.yaml:6]:
```suggestion
  label: North Carolina use tax cap
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/cap.yaml [hunk 15]
```diff
@@ -0,0 +1,138 @@
...
+  reference:
+    - title: NCDOR Use Tax
+      href: https://www.ncdor.gov/taxes-forms/sales-and-use-tax/use-tax
+    - title: 2022 NC D-400 page 2
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/cap.yaml:10]:
```suggestion
```

File: policyengine_us/parameters/gov/states/nc/tax/use_tax/rate.yaml [hunk 16]
```diff
@@ -0,0 +1,23 @@
+description:  North Carolina allows for the following use tax rate, based on adjusted gross income.
+metadata:
+  type: single_amount
+  threshold_unit: currency-USD
+  amount_unit: /1
+  label: North Carolina Use Tax rate
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/nc/tax/use_tax/rate.yaml:6]:
```suggestion
  label: North Carolina use tax rate
```

**Comment by PavelMakarchuk** [comment]
Duplicate of #4481

**Pull request closed without merging**